gpkit/_pint/__init__.py

```diff
@@ -25,13 +26,17 @@ def qty(unit):
 
 
 class GPkitUnits(object):
-    "Return monomials instead of Quantitites"
+    "Return Monomials instead of Quantitites"
 
     def __call__(self, arg):
+        "Returns a unit Monomial, caching the result for future retrievals"
         from .. import Monomial
-        return Monomial(qty(arg))
+        if arg not in MON_CACHE:
+            MON_CACHE[arg] = Monomial(qty(arg))
+        return MON_CACHE[arg]
```

Comment:
Nice. Would putting the import inside the conditional speed this up further, or is that silly?

Reply:
naw it's cached by the interpreter and basically free